File: Java.old/SpaceObject.java
package izak.asteroids;

public abstract class SpaceObject {

	public float x, y, angle;
	public float xVel, yVel, velAngle;
	
	public boolean alive = true;
	
	public abstract Vector2F[] getCurrentModel();
	
	public void updatePos() {
		x += xVel / AsteroidsComponent.TICKS_PER_SECOND;
		y += yVel / AsteroidsComponent.TICKS_PER_SECOND;
		angle += velAngle / AsteroidsComponent.TICKS_PER_SECOND;
	}
	
	public void draw(VectorGraphics vg) {
		// pass
	}
	
	public int[][] getCurrentModelInts() {
		Vector2F[] model = getCurrentModel(); 
		int[][] modelInts = new int[model.length][2];
		for (int i = 0; i < model.length; i++) {
			modelInts[i][0] = Math.round(model[i].x);
			modelInts[i][1] = Math.round(model[i].y);
		}
		return modelInts;
	}
}

File: src/components/space_object.py

from .. import constants as c

class SpaceObject:

	def __init__(self, pos_x=0, pos_y=0, angle=0):
		self.pos_x = pos_x
		self.pos_y = pos_y
		self.angle = angle
		self.vel_x, self.vel_y, self.vel_angle = 0, 0, 0
		self.alive = True

	def kill(self):
		self.alive = False

	def update_pos(self, current_time=None):
		self.pos_x += self.vel_x / c.TICKS_PER_SECOND
		self.pos_y += self.vel_y / c.TICKS_PER_SECOND
		self.angle += self.vel_angle / c.TICKS_PER_SECOND
